### PR TITLE
Added column name data class

### PR DESCRIPTION
**Pull request by radhakrishnatg**
## Fixes/Addresses/Summary/Motivation:
Adds a class to receive column names from the user. The PR has a lot of files because the metric data class is not merged yet. Please ignore the files that are part of PR #3 and review only the following files.
`data_parser/well_data_column_names.py`
`data_parser/tests/test_well_data_column_names.py`
`utils/domain_validator.py`
`utils/tests/test_domain_validator.py`

## Changes proposed in this PR:
- Added `WellDataColumnNames` class to receive column names from the user
- Added a module for different types of domain validator. Right now, it only has `InRange` domain validator. More validators will be added in the future PRs.
- Added tests for the new codes.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my
contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: primo/utils/domain_validators.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+"""
+This module contains functions that can be used for domain validation
+in Pyomo's `ConfigDict()` data structure.
+"""
+
+
+# pylint: disable-next = invalid-name
```
Comment (apd-ypuranik) [inline primo/utils/domain_validators.py:20]: Could you simply reuse in_bounds from primo/util/opt_utils?

File: primo/utils/tests/test_domain_validators.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Installed libs
+import pytest
+
+# User-defined libs
+from primo.utils.domain_validators import InRange
+
+
+def test_inrange():
+    """Tests InRange domain validator"""
```
Comment (apd-ypuranik) [inline primo/utils/tests/test_domain_validators.py:22]: This will not be needed if we can reuse the in_bounds method per comment above

**Review by apd-ypuranik — CHANGES_REQUESTED** [review]

File: primo/utils/domain_validators.py (hunk 1 above)
Comment (radhakrishnatg) [reply primo/utils/domain_validators.py:20]: The `in_bounds` function returns a `bool`. But the domain validator function must return the value if it is in bounds. E.g. If my interval is [5, 10], the `InRange` function returns the value (say 6) if it is the interval, and raises `ValueError` if it is not.

**Comment by radhakrishnatg** [comment]
@apd-ypuranik I'm not sure why the codecov test is failing. Is this expected? If not, is there something I need to address in this PR?

**Comment by apd-ypuranik** [comment]
@radhakrishnatg : It is OK to ignore the codecov test failure for now.

**Review by apd-ypuranik — APPROVED** [review]

**Review by DevKakkad32 — APPROVED** [review]

**Pull request merged**